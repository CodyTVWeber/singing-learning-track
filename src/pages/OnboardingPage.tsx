import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import { useApp } from '../context/AppContext';
import type { UserProfile } from '../models/user';
import { Button } from '../components/Button';
import { Card } from '../components/Card';
import { Container } from '../components/Container';
import { colors, fontSize, fontWeight, spacing, borderRadius } from '../theme/theme';

export const OnboardingPage: React.FC = () => {
  const navigate = useNavigate();
  const { setUser } = useApp();
  const [step, setStep] = useState(0);
  const [name, setName] = useState('');
  const [ageGroup, setAgeGroup] = useState<'kid' | 'teen' | 'adult'>('kid');

  const handleNameSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (name.trim()) {
      setStep(1);
    }
  };

  const handleComplete = async () => {
    const newUser: UserProfile = {
      id: uuidv4(),
      name: name.trim(),
      ageGroup,
      currentLevel: 1,
      totalPoints: 0,
      streak: 0,
    };

    await setUser(newUser);
    navigate('/skill-tree');
  };

  return (
    <div
      style={{
        minHeight: '100vh',
        backgroundColor: colors.background,
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        padding: spacing.md,
      }}
    >
      <Container maxWidth="500px">
        <div style={{ textAlign: 'center', marginBottom: spacing.xl }}>
          <img
            src="/img/kooka-burra-waiving.png"
            alt="Kooka the Kookaburra"
            style={{
              width: '120px',
              height: '120px',
              margin: '0 auto',
              marginBottom: spacing.lg,
            }}
          />
          <h1
            style={{
              fontSize: fontSize.xxl,
              fontWeight: fontWeight.bold,
              color: colors.primary,
              marginBottom: spacing.sm,
            }}
          >
            Welcome to Kooka Sing!
          </h1>
          <p style={{ color: colors.textLight, fontSize: fontSize.lg }}>
            Let's get to know each other
          </p>
        </div>

        {step === 0 ? (
          <Card>
            <form onSubmit={handleNameSubmit}>
              <h2
                style={{
                  fontSize: fontSize.xl,
                  fontWeight: fontWeight.semibold,
                  marginBottom: spacing.lg,
                  color: colors.darkBrown,
                }}
              >
                What's your name?
              </h2>
              <input
                type="text"
                value={name}
                onChange={(e) => setName(e.target.value)}
                placeholder="Enter your name"
                autoFocus
                style={{
                  width: '100%',
                  padding: spacing.md,
                  fontSize: fontSize.lg,
                  borderRadius: borderRadius.md,
                  border: `2px solid ${colors.featherLight}`,
                  marginBottom: spacing.lg,
                  transition: 'border-color 300ms',
                }}
                onFocus={(e) => {
                  e.currentTarget.style.borderColor = colors.primary;
                }}
                onBlur={(e) => {
                  e.currentTarget.style.borderColor = colors.featherLight;
                }}
              />
              <Button
                type="submit"
                fullWidth
                size="large"
                disabled={!name.trim()}
              >
                Continue
              </Button>
            </form>
          </Card>
        ) : (
          <Card>
            <h2
              style={{
                fontSize: fontSize.xl,
                fontWeight: fontWeight.semibold,
                marginBottom: spacing.lg,
                color: colors.darkBrown,
              }}
            >
              Hi {name}! How old are you?
            </h2>
            <div style={{ display: 'flex', flexDirection: 'column', gap: spacing.md }}>
              {[
                { value: 'kid' as const, label: 'ðŸ‘¶ Under 12', emoji: 'ðŸŽˆ' },
                { value: 'teen' as const, label: 'ðŸŽ¸ 13-17', emoji: 'ðŸŽµ' },
                { value: 'adult' as const, label: 'ðŸŽ¤ 18+', emoji: 'ðŸŽ­' },
              ].map((option) => (
                <button
                  key={option.value}
                  onClick={() => setAgeGroup(option.value)}
                  style={{
                    padding: spacing.lg,
                    borderRadius: borderRadius.lg,
                    border: `2px solid ${
                      ageGroup === option.value ? colors.primary : colors.featherLight
                    }`,
                    backgroundColor:
                      ageGroup === option.value ? colors.skyLight : colors.surface,
                    fontSize: fontSize.lg,
                    fontWeight: fontWeight.medium,
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'space-between',
                    transition: 'all 300ms',
                    cursor: 'pointer',
                  }}
                >
                  <span>{option.label}</span>
                  <span style={{ fontSize: fontSize.xxl }}>{option.emoji}</span>
                </button>
              ))}
            </div>
            <Button
              onClick={handleComplete}
              fullWidth
              size="large"
              style={{ marginTop: spacing.xl }}
            >
              Start Learning!
            </Button>
          </Card>
        )}
      </Container>
    </div>
  );
};