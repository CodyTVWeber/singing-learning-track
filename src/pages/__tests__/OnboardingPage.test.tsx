import React from 'react';
import { vi } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { OnboardingPage } from '../OnboardingPage';

vi.mock('../../context/AppContext', () => {
  return {
    useApp: () => ({
      setUser: vi.fn(async () => {}),
    }),
  };
});

describe('OnboardingPage', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <OnboardingPage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

