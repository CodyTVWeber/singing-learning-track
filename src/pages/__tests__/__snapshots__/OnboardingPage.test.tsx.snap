// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OnboardingPage > renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="min-height: 100vh; background: linear-gradient(135deg, #FAF6F2 0%, #F2E8DD 100%); display: flex; flex-direction: column; justify-content: center; padding: 16px; position: relative; overflow: hidden;"
  >
    <div
      style="position: absolute; top: -10%; left: -10%; width: 300px; height: 300px; background: linear-gradient(135deg, #4B708B 0%, #6B8CAB 100%); border-radius: 50%; filter: blur(80px); opacity: 0.3; animation: pulse 2s ease-in-out infinite; pointer-events: none; z-index: 0;"
    />
    <div
      style="position: absolute; bottom: -10%; right: -10%; width: 350px; height: 350px; background: linear-gradient(135deg, #6B5843 0%, #8B7355 50%, #D4C4B0 100%); border-radius: 50%; filter: blur(80px); opacity: 0.3; animation: pulse 2s ease-in-out infinite; animation-delay: 1s; pointer-events: none; z-index: 0;"
    />
    <div
      class=""
      style="width: 100%; max-width: 500px; margin: 0px auto; padding: 16px; position: relative; z-index: 10;"
    >
      <div
        style="text-align: center; margin-bottom: 48px; opacity: 0; transform: translateY(-20px); transition: 400ms cubic-bezier(0.4, 0, 0.2, 1);"
      >
        <img
          alt="Kooka the Kookaburra welcoming you"
          src="/img/kooka-burra-waiving.png"
          style="width: 250px; margin: 0px auto 32px auto; filter: drop-shadow(0 20px 25px -5px rgba(0, 0, 0, 0.08), 0 10px 10px -5px rgba(0, 0, 0, 0.02)); transform: scale(1); transition: transform 0.3s ease; cursor: pointer;"
        />
        <h1
          style="font-size: 48px; font-weight: 800; color: rgb(60, 60, 60); margin-bottom: 8px; letter-spacing: -0.02em;"
        >
          Welcome to Kooka Sing!
        </h1>
        <p
          style="color: rgb(107, 88, 67); font-size: 18px; font-weight: 500;"
        >
          Let's start your singing adventure together!
        </p>
      </div>
      <div
        style="opacity: 0; transform: scale(0.95); transition: all 0.4s cubic-bezier(0.4, 0, 0.2, 1);"
      >
        <div
          class=""
          style="border-radius: 20px; padding: 32px; cursor: default; transition: 400ms cubic-bezier(0.4, 0, 0.2, 1); position: relative; overflow: hidden; background-color: rgba(255, 255, 255, 0.8); box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.07), 0 2px 4px -1px rgba(0, 0, 0, 0.04); border: 1px solid rgba(255, 255, 255, 0.3);"
        >
          <div
            style="position: relative; z-index: 1;"
          >
            <form>
              <h2
                style="font-size: 32px; font-weight: 700; margin-bottom: 16px; color: rgb(60, 60, 60); text-align: center;"
              >
                What should Kooka call you?
              </h2>
              <p
                style="font-size: 16px; color: rgb(107, 88, 67); margin-bottom: 32px; text-align: center;"
              >
                Choose a name that makes you smile!
              </p>
              <input
                placeholder="Your amazing name"
                style="width: 100%; padding: 16px 24px; font-size: 18px; border-radius: 9999px; border: 2px solid rgb(238, 238, 232); background-color: rgb(255, 255, 255); margin-bottom: 32px; transition: 400ms cubic-bezier(0.4, 0, 0.2, 1); text-align: center; font-weight: 500; border-color: rgb(139, 115, 85); box-shadow: 0 0 0 4px #A68B6820;"
                type="text"
                value=""
              />
              <button
                disabled=""
                style="padding: 16px 32px; border-radius: 9999px; font-size: 18px; font-weight: 600; transition: none; display: inline-flex; align-items: center; justify-content: center; gap: 8px; width: 100%; min-height: 56px; cursor: not-allowed; opacity: 0.6; position: relative; overflow: hidden; text-decoration: none; white-space: nowrap; user-select: none; flex-direction: row-reverse; background: linear-gradient(135deg, #8B7355 0%, #A68B68 100%); color: rgb(250, 246, 242); box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.08), 0 4px 6px -2px rgba(0, 0, 0, 0.02);"
                type="submit"
              >
                <span
                  style="display: flex; align-items: center;"
                >
                  <svg
                    aria-label="forward icon"
                    class=""
                    fill="currentColor"
                    style="width: 24px; height: 24px; color: rgb(60, 60, 60); cursor: default; transition: none;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </span>
                Let's Go!
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        style="position: absolute; top: 15%; left: 5%; animation: float 5s ease-in-out infinite; opacity: 0.2; pointer-events: none; z-index: 1;"
      >
        <img
          alt="Floating Kooka"
          src="/img/kooka-burra-flying.png"
          style="width: 60px; border-radius: 50%; filter: drop-shadow(0 2px 4px -1px rgba(0, 0, 0, 0.06), 0 1px 2px -1px rgba(0, 0, 0, 0.04));"
        />
      </div>
      <div
        style="position: absolute; bottom: 20%; right: 5%; animation: float 6s ease-in-out infinite; animation-delay: 2s; opacity: 0.2; pointer-events: none; z-index: 1;"
      >
        <img
          alt="Floating Kooka"
          src="/img/kooka-burra-singing.png"
          style="width: 50px; border-radius: 50%; filter: drop-shadow(0 2px 4px -1px rgba(0, 0, 0, 0.06), 0 1px 2px -1px rgba(0, 0, 0, 0.04));"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
